Makefile

```diff
@@ -13,29 +13,90 @@
 # limitations under the License.
 #
 
-PKG=github.com/kubernetes-sigs/aws-efs-csi-driver
-IMAGE?=amazon/aws-efs-csi-driver
 VERSION=v1.3.8
+
+PKG=github.com/kubernetes-sigs/aws-efs-csi-driver
 GIT_COMMIT?=$(shell git rev-parse HEAD)
 BUILD_DATE?=$(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
+
 EFS_CLIENT_SOURCE?=k8s
-IMAGE_PLATFORMS?=linux/arm64,linux/amd64
 LDFLAGS?="-X ${PKG}/pkg/driver.driverVersion=${VERSION} \
 		  -X ${PKG}/pkg/driver.gitCommit=${GIT_COMMIT} \
 		  -X ${PKG}/pkg/driver.buildDate=${BUILD_DATE} \
 		  -X ${PKG}/pkg/driver.efsClientSource=${EFS_CLIENT_SOURCE}"
+
 GO111MODULE=on
 GOPROXY=direct
 GOPATH=$(shell go env GOPATH)
 GOOS=$(shell go env GOOS)
 
+REGISTRY?=amazon
+IMAGE?=$(REGISTRY)/aws-efs-csi-driver
+TAG?=$(GIT_COMMIT)
+
+OUTPUT_TYPE?=docker
+
+OS?=linux
+ARCH?=amd64
+OSVERSION?=amazon
+
+ALL_OS?=linux
+ALL_ARCH_linux?=amd64 arm64
+ALL_OSVERSION_linux?=amazon
+ALL_OS_ARCH_OSVERSION_linux=$(foreach arch, $(ALL_ARCH_linux), $(foreach osversion, ${ALL_OSVERSION_linux}, linux-$(arch)-${osversion}))
+
+ALL_OS_ARCH_OSVERSION=$(foreach os, $(ALL_OS), ${ALL_OS_ARCH_OSVERSION_${os}})
+
+PLATFORM?=linux/amd64,linux/arm64
+
+# split words on hyphen, access by 1-index
+word-hyphen = $(word $2,$(subst -, ,$1))
+
 .EXPORT_ALL_VARIABLES:
 
-.PHONY: aws-efs-csi-driver
-aws-efs-csi-driver:
-	mkdir -p bin
-	@echo GOARCH:${GOARCH}
-	CGO_ENABLED=0 GOOS=linux go build -mod=vendor -ldflags ${LDFLAGS} -o bin/aws-efs-csi-driver ./cmd/
+.PHONY: linux/$(ARCH) bin/aws-efs-csi-driver
+linux/$(ARCH): bin/aws-efs-csi-driver
+bin/aws-efs-csi-driver: | bin
+	CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) go build -mod=vendor -ldflags ${LDFLAGS} -o bin/aws-efs-csi-driver ./cmd/
+
+.PHONY: all
+all: all-image-docker
+
+.PHONY: all-push
+all-push:
+	docker buildx build \
+		--no-cache-filter=linux-amazon \
```

Comment:
nice to build the no cache! would this be helpful on avoiding unnecessary image cache triggered CVEs?

Reply:
yes, this is to avoid case where somehow you have cache with older yum packages

Reply:
nice, I will give it a build and push to my dev account to see this can help sort out most of our CVE mitigation